presto-tests/src/test/java/com/facebook/presto/execution/TestEventListener.java

```diff
@@ -168,6 +175,61 @@ public void testNormalQuery()
         assertEquals(queryCompletedEvent.getStatistics().getTotalRows(), expectedCompletedPositions);
     }
 
+    @Test
+    public void testPrepareAndExecute()
+            throws Exception
+    {
+        final String selectQuery = "SELECT count(*) FROM lineitem WHERE shipmode = ?";
+        final String prepareQuery = "PREPARE stmt1 FROM " + selectQuery;
+
+        // QueryCreated: 1, QueryCompleted: 1, Splits: 0
+        runQueryAndWaitForEvents(prepareQuery, 2);
+
+        QueryCreatedEvent queryCreatedEvent = getOnlyElement(generatedEvents.getQueryCreatedEvents());
+        assertEquals(queryCreatedEvent.getContext().getServerVersion(), "testversion");
+        assertEquals(queryCreatedEvent.getContext().getServerAddress(), "127.0.0.1");
+        assertEquals(queryCreatedEvent.getContext().getEnvironment(), "testing");
+        assertEquals(queryCreatedEvent.getContext().getClientInfo().get(), "{\"clientVersion\":\"testVersion\"}");
+        assertEquals(queryCreatedEvent.getMetadata().getQuery(), prepareQuery);
+
+        QueryCompletedEvent queryCompletedEvent = getOnlyElement(generatedEvents.getQueryCompletedEvents());
+        assertTrue(queryCompletedEvent.getContext().getResourceGroupId().isPresent());
+        assertEquals(queryCompletedEvent.getContext().getResourceGroupId().get(), createResourceGroupId("global", "user-user"));
+        assertEquals(queryCompletedEvent.getIoMetadata().getOutput(), Optional.empty());
+        assertEquals(queryCompletedEvent.getIoMetadata().getInputs().size(), 0);  // Prepare has no inputs
+        assertEquals(queryCompletedEvent.getContext().getClientInfo().get(), "{\"clientVersion\":\"testVersion\"}");
+        assertEquals(queryCreatedEvent.getMetadata().getQueryId(), queryCompletedEvent.getMetadata().getQueryId());
+        assertFalse(queryCompletedEvent.getMetadata().getPrepareQuery().isPresent());
```

Comment:
Also add this assertion to the other tests (since they shouldn't have a prepare query)